libs/kotlinlib/src/mill/kotlinlib/ksp/KspModule.scala

```diff
@@ -360,6 +390,71 @@ trait KspModule extends KotlinModule { outer =>
 
   }
 
+  def kspWorkerModule: ModuleRef[KspWorkerModule] = ModuleRef(KspWorkerModule)
+
+  /**
+   * The Kotlin compile task with KSP.
+   * This task should run as part of the [[generatedSources]] task to
+   * so that the generated  sources are in the [[compileClasspath]]
+   * for the main compile task. It uses an in-process worker to run KSP
+   * provided from [[kspWorkerModule]]
+   */
+  private def generatedSourcesWithKsp2: T[GeneratedKspSources] = Task {
+
+    val processorResolvedClasspath = kotlinSymbolProcessorsResolved().map(_.path)
+    val processorClasspath = processorResolvedClasspath.mkString(File.pathSeparator)
+
+    val kspOutputDir = Task.dest / "generated"
+    val java = kspOutputDir / "java"
+    val kotlin = kspOutputDir / "kotlin"
+    val resources = kspOutputDir / "resources"
+    val classes = kspOutputDir / "classes"
+    val kspCachesDir = Task.dest / "caches"
+
+    val processorOptionsValue =
+      kspProcessorOptions().map((key, value) => s"$key=$value").toSeq.mkString(File.pathSeparator)
+
+    val processorOptions = if (processorOptionsValue.isEmpty)
+      ""
+    else
+      s"-processor-options=${processorOptionsValue}"
+    val args = Seq(
+      s"-module-name=${kspModuleName}",
+      "-jvm-target",
+      kspJvmTarget(),
+      s"-jdk-home=${System.getProperty("java.home")}",
+      s"-source-roots=${kspSources().map(_.path).mkString(File.pathSeparator)}",
+      s"-project-base-dir=${moduleDir.toString}",
+      s"-output-base-dir=${kspOutputDir}",
+      s"-caches-dir=${kspCachesDir}",
+      s"-libraries=${kspClasspath().map(_.path).mkString(File.pathSeparator)}",
+      s"-class-output-dir=${classes}",
+      s"-kotlin-output-dir=${kotlin}",
+      s"-java-output-dir=${java}",
+      s"-resource-output-dir=${resources}",
+      s"-language-version=${kspLanguageVersion()}",
+      s"-incremental=true",
+      s"-incremental-log=true",
+      s"-api-version=${kspApiVersion()}",
+      processorOptions,
+      s"-map-annotation-arguments-in-java=false"
+    ) ++ ksp2Args() :+ processorClasspath
```

Comment:
Need to clean up here, this classpath is not needed as we load the whole thing in the classloader (KSP tools deps + processor classpath)